tests/test_start_date.py

```diff
@@ -0,0 +1,460 @@
+
+        state_2 = menagerie.get_state(conn_id)
+
+        for stream in self.testable_streams():
+            with self.subTest(stream=stream):
+                replication_type = self.expected_replication_method().get(stream)
+                record_count_1 = record_count_by_stream_1.get(stream, 0)
+                record_count_2 = record_count_by_stream_2.get(stream, 0)
+
+                # Testing how FULL TABLE streams handle start date
+                if replication_type == self.FULL:
+
+                    # Verify that a bookmark doesn't exist for the stream.
+                    self.assertTrue(state_1.get(stream) is None,
+                                    msg="There should not be bookmark value for {}\n{}".format(stream, state_1.get(stream)))
+                    self.assertTrue(state_2.get(stream) is None,
+                                    msg="There should not be bookmark value for {}\n{}".format(stream, state_1.get(stream)))
+
+                    # Verify that the 2nd sync includes the same number of records as the 1st sync.
+                    # -> Currently full table does not obey start_date, which makes this assertion valid
+                    self.assertEqual(record_count_2, record_count_1,
+                                     msg="\nStream '{}' is {}\n".format(stream, self.FULL) +
+                                     "Record counts should be equal, but are not\n" +
+                                     "Sync 1 start_date: {} ".format(start_date_1) +
+                                     "Sync 1 record_count: {}\n".format(record_count_1) +
+                                     "Sync 2 start_date: {} ".format(start_date_2) +
+                                     "Sync 2 record_count: {}".format(record_count_2))
+
+
+                    # Verify all records in the 1st sync are included in the 2nd sync since
+                    # 2nd sync has a later start date.
+                    records_from_sync_1 = set(row.get('data').get('eid')
+                                              for row in synced_records_1.get(stream, []).get('messages', []))
+                    records_from_sync_2 = set(row.get('data').get('eid')
+                                              for row in synced_records_2.get(stream, []).get('messages', []))
+                    self.assertEqual(set(), records_from_sync_1.difference(records_from_sync_2),
+                                     msg="Sync 2 record(s) missing from Sync 1:\n{}".format(
+                                         records_from_sync_2.difference(records_from_sync_1))
+                    )
+
+                else:
+                    raise Exception("Expectations are set incorrectly. {} cannot have a "
+                                    "replication method of {}".format(stream, replication_type))
+
+        # TODO Remove when test complete
+        print("\n\n\tTOOD's PRESENT | The test is incomplete\n\n")
+
+class TestSquareStartDateIncremental(TestSquareBase):
```

Comment:
Don't want to typically break this up. Might be worth it though. Only did for adroll because of limitations with creating data for incremental streams.

Reply:
Merged 👍